src/graph/test/SchemaTest.cpp

```diff
@@ -1376,5 +1376,110 @@ TEST_F(SchemaTest, TTLtest) {
     }
 }
 
+TEST_F(SchemaTest, addSchemaFromSpaceTest) {
+    auto client = gEnv->getClient();
+    ASSERT_NE(nullptr, client);
+    // prepare
+    cpp2::ExecutionResponse resp;
+    std::string cmd = "CREATE SPACE schema_space(partition_num=9, replica_factor=1)";
+    auto code = client->execute(cmd, resp);
+    ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+    cmd = "USE schema_space";
+    code = client->execute(cmd, resp);
+    ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+    cmd = "CREATE TAG tag_1(name string); CREATE EDGE edge_1(name string);";
+    code = client->execute(cmd, resp);
+    ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+    // add schemas into empty space
+    {
+        cmd = "CREATE SPACE empty_space(partition_num=9, replica_factor=1)";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        cmd = "USE empty_space";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        cmd = "ADD SCHEMA FROM schema_space;";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        cmd = "SHOW TAGS;";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        std::vector<std::tuple<std::string>> expected1{
+                {"tag_1"},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected1, true, {0}));
+
+        cmd = "SHOW EDGES;";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        std::vector<std::tuple<std::string>> expected2{
+                {"edge_1"},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected2, true, {0}));
+    }
+    // add schemas into the same name space
+    {
+        cmd = "USE schema_space";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+
+        cmd = "ADD SCHEMA FROM schema_space;";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code);
+    }
+    // add schemas into space which has existent tag
+    {
+        cmd = "ADD SCHEMA FROM schema_space;";
+        code = client->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::E_EXECUTION_ERROR, code);
+    }
+    // add schemas into not empty space
+    {
+        cmd = "CREATE SPACE space_9(partition_num=9, replica_factor=1)";
```

Comment:
What if the target space shares different partition_num and replica_factor from the target space? Will schema be added successfully? Moreover, what if the source space contains an index? Will index be also copied?

Reply:
Yeah, the default value also is copied. The configuration of different spaces is not affect.